conversion_proxy/src/lib.rs

```diff
@@ -443,61 +574,80 @@ mod tests {
-        testing_env!(context);
-        let (to, _amount, _fee_address, _fee_amount, _max_rate_timespan) = default_values();
-        contract.set_oracle_account(to);
+        contract.set_feed_payer();
+        assert_eq!(contract.get_feed_payer().to_vec(), env::signer_account_pk());
     }
 
     #[test]
     #[should_panic(expected = r#"ERR_PERMISSION"#)]
-    fn admin_provider_no_permission() {
-        let context = get_context(alice_account(), ntoy(1), 10u64.pow(14), false);
-        testing_env!(context);
+    fn admin_feed_parser_no_permission() {
+        testing_env!(get_context(alice_account(), ntoy(1), 10u64.pow(14), false));
         let mut contract = ConversionProxy::default();
-        let (to, _amount, _fee_address, _fee_amount, _max_rate_timespan) = default_values();
-        contract.set_provider_account(to);
+        let (to, _, _, _, _) = default_values();
+        contract.set_feed_parser(to.into());
     }
 
     #[test]
-    fn admin_provider() {
+    fn admin_feed_parser() {
         let owner = ConversionProxy::default().owner_id;
         let mut contract = ConversionProxy::default();
-        let context = get_context(owner, ntoy(1), 10u64.pow(14), false);
-        testing_env!(context);
-        let (to, _amount, _fee_address, _fee_amount, _max_rate_timespan) = default_values();
-        contract.set_provider_account(to);
+        testing_env!(get_context(owner, ntoy(1), 10u64.pow(14), false));
+        let (to, _, _, _, _) = default_values();
+        contract.set_feed_parser(to.clone().into());
+        assert_eq!(contract.get_feed_parser(), Into::<AccountId>::into(to));
     }
 
     #[test]
     #[should_panic(expected = r#"ERR_PERMISSION"#)]
     fn admin_owner_no_permission() {
-        let context = get_context(alice_account(), ntoy(1), 10u64.pow(14), false);
-        testing_env!(context);
+        testing_env!(get_context(alice_account(), ntoy(1), 10u64.pow(14), false));
         let mut contract = ConversionProxy::default();
-        let (to, _amount, _fee_address, _fee_amount, _max_rate_timespan) = default_values();
+        let (to, _, _, _, _) = default_values();
         contract.set_owner(to);
     }
 
     #[test]
     fn admin_owner() {
         let owner = ConversionProxy::default().owner_id;
         let mut contract = ConversionProxy::default();
-        let context = get_context(owner, ntoy(1), 10u64.pow(14), false);
-        testing_env!(context);
-        let (to, _amount, _fee_address, _fee_amount, _max_rate_timespan) = default_values();
-        contract.set_owner(to);
+        testing_env!(get_context(owner, ntoy(1), 10u64.pow(14), false));
+        let (to, _, _, _, _) = default_values();
+        contract.set_owner(to.clone());
+        testing_env!(get_context(to.into(), ntoy(1), 10u64.pow(14), false));
+        assert!(contract.owner_id == env::signer_account_id());
+        assert!(contract.get_feed_payer().to_vec() != env::signer_account_pk());
+        contract.set_feed_payer();
```

Comment:
If the `owner` is always the `feed_payer` then it seems odd to me that the `set_feed_payer` method exists and must be called separately from the `set_owner` method.

Reply:
I couldn't find a way to get the pk from the accountId and it makes sense that it's not possible so it's the way I found.

Reply:
Okay :+1: